stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -2332,8 +2337,8 @@ function getindex_I_sorted_linear(A::AbstractSparseMatrixCSC{Tv,Ti}, I::Abstract
                 ptrA += 1
             elseif rowI < rowA
                 ptrI += 1
```

Comment:
Couldn't one do a binary search here?  Aren't the row indices sorted?

Reply:
yes that is possible. It comes at the cost of a log(k) factor where k is the size of I. I was hesitating between the binary search and the hashmap here as I understand `getindex_I_sorted_linear`, to be the fall-back when both the column in A has relatively many nonzero entries and I is fairly large. So the idea is to go through both I and A[:,j] simultaneously. It seemed to me that if one had an efficient hashmap, this would be the algorithm of choice in this situation.

Reply:
a hash map has a "hidden cost" of `log(k)` (or worse!) as well, so it is not a priori the better choice IMHO, one should benchmark.

Reply:
accessing it. Hash map is `O(1)` just because the memory is finite (but then, essentially everything is `O(1)`). So there's no real theoretical advantage. It should be benchmarked in my opinion.

Reply:
I see .. I believe the right thing to do for this algorithm might be to go with a binary search in `I` as @stevengj suggested. This still avoids the cache that is linear in `n` and should be considerably faster than the current hashmap implementation I proposed. However, for small matrices it will be slower than the `O(1)` access of the linear cache... nothing beats that for small matrices

Reply:
(I will implement it when I have some time)